Playmate/View Controllers/Session Details View/PlayerProfileViewController.m

```diff
@@ -124,54 +124,50 @@ - (IBAction)didTapFriend:(id)sender {
     
     if (self.isMyFriend) {
         // if removing friend
-        
-        PFUser *me = [[PFUser currentUser] fetchIfNeeded];
-        
         // remove self.user.objectId from my friends list
-        PlayerConnection *myPc = [me[@"playerConnection"][0] fetchIfNeeded];
+        PlayerConnection *myPlayerConnection = [user[@"playerConnection"][0] fetchIfNeeded];
         
-        NSMutableArray *tempFriendsList = (NSMutableArray *)myPc.friendsList;
+        NSMutableArray *tempFriendsList = (NSMutableArray *)myPlayerConnection.friendsList;
         [tempFriendsList removeObject:self.user.objectId];
-        myPc.friendsList = (NSArray *)tempFriendsList;
+        myPlayerConnection.friendsList = (NSArray *)tempFriendsList;
         
-        [myPc saveInBackground];
+        [myPlayerConnection saveInBackground];
         
-        // remove me.objectId from self.user.pc friends list
-        PlayerConnection *theirPc = [self.user[@"playerConnection"][0] fetchIfNeeded];
+        // remove user.objectId from self.user.playerconnect friends list
+        PlayerConnection *theirPlayerConnection = [self.user[@"playerConnection"][0] fetchIfNeeded];
         
-        tempFriendsList = (NSMutableArray *)theirPc[@"friendsList"];
-        [tempFriendsList removeObject:me.objectId];
-        theirPc[@"friendsList"] = (NSArray *)tempFriendsList;
+        tempFriendsList = (NSMutableArray *)theirPlayerConnection[@"friendsList"];
```

Comment:
Love the renaming. 

Still one question. 

I see in line 128 and in line 137
`PlayerConnection *myPlayerConnection = [user[@"playerConnection"][0] fetchIfNeeded];`
`PlayerConnection *theirPlayerConnection = [self.user[@"playerConnection"][0] fetchIfNeeded];`

what is the difference between `myPlayerConnection` and `theirPlayerConnection`

Reply:
Ah, my PlayerConnection model is used to represent my friend network and every PFUser(player) has their own PlayerConnection object connected to the PFUser's objectId. My friends network an undirected graph and to store data in parse, I use an adjacency list representation such that if Player A and Player B are friends, then B is in A's friend list and A is in B's friend list. That is why there is myPlayerConnection (let's say I am player A, then I need my PC to add B to the friend list) and theirPlayerConnection (to add me, player A, to Player B's friend list).